syncstorage/storage/sql/queries_spanner.py

```diff
@@ -0,0 +1,114 @@
+from sqlalchemy.sql import select, bindparam
+
+DATABASE_CREATE_DDL = """\
+CREATE TABLE user_collections (
+    userid          STRING(MAX) NOT NULL,
+    collection      INT64 NOT NULL,
+    last_modified   TIMESTAMP NOT NULL
+) PRIMARY KEY (userid, collection);
+
+CREATE TABLE bso (
+    userid          STRING(MAX) NOT NULL,
+    collection      INT64 NOT NULL,
+    id              STRING(MAX) NOT NULL,
+    sortindex       INT64,
+    modified        TIMESTAMP NOT NULL,
+    payload         STRING(MAX) NOT NULL,
+    ttl             TIMESTAMP NOT NULL
+) PRIMARY KEY (userid, collection, id),
+  INTERLEAVE IN PARENT user_collections ON DELETE CASCADE;
+
+CREATE INDEX BsoTtl ON bso(ttl);
+CREATE INDEX BsoLastModified ON bso(userid, collection, modified);
+
+CREATE TABLE collections (
+    collectionid    INT64 NOT NULL,
+    name            STRING(MAX) NOT NULL,
+) PRIMARY KEY (collectionid);
+
+CREATE TABLE batches (
+    userid          STRING(MAX) NOT NULL,
+    collection      INT64 NOT NULL,
+    id              TIMESTAMP NOT NULL,
+    bsos            STRING(MAX) NOT NULL,
+    expiry          TIMESTAMP NOT NULL
+) PRIMARY KEY (userid, collection, id);
```

Comment:
IIUC, this needs an `ON DELETE CASCADE` in order for this earlier comment to be true:

```
# Deleting user_collections will wipe out all BSO's and batches
# for the user due to cascading delete
```

Reply:
we're just going to remove this comment, it's outdated. AFAICT the original impl. doesn't actually delete batches when deleting storage so we match its behavior.

we originally interleaved these tables which would have caused the cascade but removed it -- i've added this comment to the DDL:
```
-- batches is not interleaved w/ user_collections because batches can
-- be created before there's a row in that table (test_storage has a
-- test that triggers this). spanner requires the parent's row to
-- be present before the child's
```